cmd/fpga_crihook/main.go

```diff
@@ -140,65 +225,19 @@ func (he *hookEnv) getFPGAParams(content map[string]interface{}) (*fpgaParams, e
 
 }
 
-func (he *hookEnv) validateBitStream(params *fpgaParams, fpgaBitStreamPath string) error {
-	output, err := he.execer.Command("packager", "gbs-info", "--gbs", fpgaBitStreamPath).CombinedOutput()
-	if err != nil {
-		return errors.Wrapf(err, "%s/%s: can't get bitstream info", params.region, params.afu)
-	}
-
-	reader := bytes.NewBuffer(output)
-	content, err := decodeJSONStream(reader)
-	if err != nil {
-		return errors.WithMessage(err, fmt.Sprintf("%s/%s: can't decode 'packager gbs-info' output", params.region, params.afu))
-	}
-
-	afuImage, ok := content["afu-image"]
-	if !ok {
-		return errors.Errorf("%s/%s: 'afu-image' field not found in the 'packager gbs-info' output", params.region, params.afu)
-	}
-
-	interfaceUUID, ok := afuImage.(map[string]interface{})["interface-uuid"]
-	if !ok {
-		return errors.Errorf("%s/%s: 'interface-uuid' field not found in the 'packager gbs-info' output", params.region, params.afu)
-	}
-
-	acceleratorClusters, ok := afuImage.(map[string]interface{})["accelerator-clusters"]
-	if !ok {
-		return errors.Errorf("%s/%s: 'accelerator-clusters' field not found in the 'packager gbs-info' output", params.region, params.afu)
-	}
-
-	if canonize(interfaceUUID.(string)) != params.region {
-		return errors.Errorf("bitstream is not for this device: region(%s) and interface-uuid(%s) don't match", params.region, interfaceUUID)
-	}
-
-	acceleratorTypeUUID, ok := acceleratorClusters.([]interface{})[0].(map[string]interface{})["accelerator-type-uuid"]
-	if !ok {
-		return errors.Errorf("%s/%s: 'accelerator-type-uuid' field not found in the 'packager gbs-info' output", params.region, params.afu)
-	}
-
-	if canonize(acceleratorTypeUUID.(string)) != params.afu {
-		return errors.Errorf("incorrect bitstream: AFU(%s) and accelerator-type-uuid(%s) don't match", params.afu, acceleratorTypeUUID)
-	}
-
-	return nil
-}
-
-func (he *hookEnv) programBitStream(params *fpgaParams, fpgaBitStreamPath string) error {
-	output, err := he.execer.Command("fpgaconf", "-S", params.devNum, fpgaBitStreamPath).CombinedOutput()
-	if err != nil {
-		return errors.Wrapf(err, "failed to program AFU %s to socket %s, region %s: output: %s", params.afu, params.devNum, params.region, string(output))
-	}
-
-	programmedAfu, err := he.getProgrammedAfu(params.devNum)
-	if err != nil {
-		return err
-	}
-
-	if programmedAfu != params.afu {
-		return errors.Errorf("programmed function %s instead of %s", programmedAfu, params.afu)
+func (he *hookEnv) getBitStream(params *fpgaParams) (fpgaBitStream, error) {
+	bitStreamPath := ""
+	for _, ext := range []string{".gbs", ".aocx"} {
+		bitStreamPath = filepath.Join(he.bitStreamDir, params.region, params.afu+ext)
+		if _, err := os.Stat(bitStreamPath); !os.IsNotExist(err) {
```

Comment:
What if `err` is not `ENOENT`, but `EBUSY` or `ELOOP`? Should we create bit stream objects in this case?